disco_aws_automation/disco_elasticsearch.py

```diff
@@ -0,0 +1,404 @@
+        value = self.get_endpoint(domain_name)
+
+        return self.route53.delete_records_by_value('CNAME', value)
+
+    def _describe_es_domain(self, domain_name):
+        """
+        Returns domain configuration information about the specified
+        Elasticsearch domain, including the domain ID, domain endpoint, and
+        domain ARN.
+        """
+        return self.conn.describe_elasticsearch_domain(DomainName=domain_name)
+
+    def get_endpoint(self, domain_name):
+        """
+        Get Elasticsearch service endpoint
+        """
+        try:
+            return self._describe_es_domain(domain_name)['DomainStatus']['Endpoint']
+        except (BotoCoreError, Boto3Error, KeyError):
+            return None
+
+    def _access_policy(self, domain_name, allowed_source_ips):
+        """
+        Construct an access policy for the new Elasticsearch cluster. Needs to be dynamically created because
+        it will use the environment's proxy hostclass to forward requests to the elasticsearch cluster and the
+        IP of the proxy hostclass could be different at run time.
+        """
+        proxy_hostclass = self.get_aws_option('http_proxy_hostclass')
+        proxy_ip = self.get_hostclass_option('eip', proxy_hostclass)
+
+        allowed_source_ips.append(proxy_ip)
+
+        resource = "arn:aws:es:{region}:{account}:domain/{domain_name}/*".format(region=self.region,
+                                                                                 account=self.account_id,
+                                                                                 domain_name=domain_name)
+
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Effect": "Allow",
+                    "Principal": {
+                        "AWS": "*"
+                    },
+                    "Action": "es:*",
+                    "Resource": resource,
+                    "Condition": {
+                        "IpAddress": {
+                            "aws:SourceIp": allowed_source_ips
+                        }
+                    }
+                }
+            ]
+        }
+        return json.dumps(policy)
+
+    def create(self, elasticsearch_name=None, es_config=None):
+        """
+        Create an ElasticSearch domain.
+        """
+        if elasticsearch_name:
+            desired_elasticsearch_names = [elasticsearch_name]
+        else:
+            desired_elasticsearch_names = self._get_elasticsearch_names()
+
+        all_elasticsearch_names = [domain_info["internal_name"] for domain_info in self.list()]
+
+        for elasticsearch_name in desired_elasticsearch_names:
+            domain_name = self.get_domain_name(elasticsearch_name)
+            if elasticsearch_name in all_elasticsearch_names:
+                logging.info('ElasticSearch domain %s already exists. Try updating it instead.', domain_name)
```

Comment:
Arguably we could combine create with update so that disco_elasticsearch.py would be more in line with how we've handled elasticache and rds. Would that make sense?